refactor client swapping script

markgoho opened the pull request:
re-opening this PR on behalf of @ZASMan 

this PR needs to be reviewed thoroughly by
- [ ] @TreyE 
- [ ] @saimekala07 

and optionally
- [ ] @matt--williams 

This affects how all developers swap clients locally and how we swap clients in our docker images

More details here: MAINT-180558816

ZASMan committed:
Refactor resource registry to read from config folders and move less files.

Remove root level system folder.

Fixes some stuff.

stuff

stuff

stuff

stuff

ZASMan committed:
Remove root directory system folder.

ZASMan commented:
> re-opening this PR on behalf of @ZASMan
> 
> this PR needs to be reviewed thoroughly by
> 
> * [ ]  @TreyE
> * [ ]  @saimekala07
> 
> and optionally
> 
> * [ ]  @matt--williams
> 
> This affects how all developers swap clients locally and how we swap clients in our docker images
> 
> More details here: [MAINT-180558816](https://www.pivotaltracker.com/n/projects/2481179/stories/180558816)

Alright, all this PR is doing is removing the need for a `system/` folder held in the root directory of Enroll app. So instead of moving a bunch of folders over into the root directory, there's a file called `client_configuration.txt` that is modified by the rake task. 

From a developer process perspective, developers now only need to change the files in the `config/client_config/#{state_name}/` folder.

From a devops perspective, the `deploy processes are exactly the same`, you'll just see less files changed. Basically only `settings.yml` and a few random stylesheets will change I believe.

The next step would be to double check those few files and also refactor out `settings.yml`, which I've outlined here: MAINT-178128930

ZASMan committed:
Merge branch 'trunk' into 180558816-resource-registry-refactor

bill-transue committed:
DRY up client differences to support ENV client

This DRYs up the client differences to support the goal of one Docker
image for every environment.

bill-transue committed:
Rename scss to erb

bill-transue committed:
Merge remote-tracking branch 'origin/trunk' into 180558816-resource-registry-refactor
